### PR TITLE
Add gcm android push notifications

### PR DESCRIPTION
**Pull request by lionleaf**
Server er på denne branchen. Kjører purq fordi jeg vil ha code review. Dette var hacket sammen.

### CODE REVIEW

**Comment by hermansc** [comment]
Hihi, alright, you asked for it ;-)

**Comment by lionleaf** [comment]
Thanks :) I'll fix it all later today.

File: routes/door.js [hunk 1]
```diff
@@ -53,3 +55,6 @@ exports.closed = function (req, res) {
     })
   });
 }
+
+
+
```
Comment (cristeahub) [inline routes/door.js]: These lines are not needed

File: routes/gcm.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    http = require('http'), 
+    url = require('url'),
+    qs = require('querystring'),
+    Server = mongo.Server,
+    Db = mongo.Db,
+    BSON = mongo.BSONPure;
+
+var server = new Server('localhost', 27017, {auto_reconnect: true});
+db = new Db('hackerspacedb', server);
+
```
Comment (cristeahub) [inline routes/gcm.js:14]: one line split while two are used later down

File: routes/gcm.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    delayWhileIdle: false,
+    timeToLive: 3,
+    data: {
+      message: msg
+    } 
+  });
+
+  var sender = new gcm.Sender(config.gcm.sender);
+
+
```
Comment (cristeahub) [inline routes/gcm.js]: woah, there's so much room for activities :smile: hehe ^^

File: server.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
+, gcm = require('./routes/gcm.js')
+, app = express();
 
 app.configure(function () {
-  app.use(express.logger('dev')); /* 'default', 'short', 'tiny', 'dev' */
-  app.use(express.bodyParser());
-});
+    app.use(express.logger('dev')); /* 'default', 'short', 'tiny', 'dev' */
+    app.use(express.bodyParser());
+    });
```
Comment (cristeahub) [inline server.js]: this line should not be indented

**Comment by cristeahub** [comment]
Suggestion: rebase the new changes into existing commits instead of adding more commits.
I.e. change the error you introduced instead of patching it later.

**Comment by lionleaf** [comment]
Alright, rebased and fixed what you found :) Anything else?

File: server.js [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
 });
 
+app.post('/register-android', gcm.register);
 app.get('/door', door.isOpen);
 app.post('/door', door.opened);
 app.put('/door', door.closed);
 
 app.listen(3000);
 console.log('Listen on port 3000...')
+
```
Comment (cristeahub) [inline server.js:19]: This line is unessecary

File: server.js [hunk 6]
```diff
@@ -1,15 +1,18 @@
 var express = require('express')
-  , door = require('./routes/door.js')
-  , app = express();
+, door = require('./routes/door.js')
+, gcm = require('./routes/gcm.js')
+, app = express();
```
Comment (cristeahub) [inline server.js:4]: indent these lines

File: routes/gcm.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        // Params: message-literal, registrationIds-array, No. of retries, callback-function
+        sender.send(message, registrationIds[i], 4, function (err, result) {
+        console.log(result);
+        });
+      }   
+    });
+
+ });
+}
+
```
Comment (cristeahub) [inline routes/gcm.js:99]: No need for the extra line here

File: routes/door.js [hunk 8]
```diff
@@ -53,3 +55,4 @@ exports.closed = function (req, res) {
     })
   });
 }
+
```
Comment (cristeahub) [inline routes/door.js:58]: This line is not needed

**Pull request merged**